ui/app/rif/pages/domains/register/index.js

```diff
@@ -0,0 +1,375 @@
+        ...this.props,
+        selectedAddress: address,
+      })
+    }
+  }
+
+  // TODO: change this to use obs store
+  storePendingDomain () {
+    let domains = JSON.parse(localStorage.getItem('rnsDomains'));
+    const expirationDate = new Date();
+    expirationDate.setFullYear(expirationDate.getFullYear() + this.props.yearsToRegister);
+    const pendingDomain = {
+      domain: this.props.domainName,
+      expiration: expirationDate.toDateString(),
+      autoRenew: false,
+      status: 'pending',
+      address: this.props.selectedAddress,
+      content: '',
+      ownerAddress: this.props.selectedAddress,
+      isLuminoNode: true,
+      isRifStorage: true,
+      resolvers: [],
+    };
+    if (!domains) {
+      domains = [];
+    }
+    if (domains.filter(domain => domain.domain === this.props.domainName).length <= 0) {
+      domains.push(pendingDomain);
+    }
+    localStorage.setItem('rnsDomains', JSON.stringify(domains));
+  }
+
+  changeDomainStatusAndStore () {
+    const domains = JSON.parse(localStorage.getItem('rnsDomains'));
+    const domain = domains.filter(domain => domain.domain === this.props.domainName)[0];
+    domain.status = 'active';
+    localStorage.setItem('rnsDomains', JSON.stringify(domains));
+  }
+  // TODO: change this to use obs store
+
+  showRegistration () {
+    this.props.getCost(this.props.domainName, 1)
+      .then(costInWei => {
+        const costInRif = costInWei / 1e18;
+        this.props.showThis({
+          ...this.props,
+          currentStep: 'register',
+          yearsToRegister: 1,
+          costInRif: costInRif,
+          gasCost: 0,
+        });
+      });
+  }
+
+  async initiateRegistration () {
+    const commitment = await this.props.requestRegistration(this.props.domainName, this.props.yearsToRegister);
+    await this.props.wait();
+    this.showConfirmTransactionPage((commitment) => {
+      this.storePendingDomain();
+      this.afterCommitSubmission(commitment)
+    }, commitment);
+  }
+
+  async completeRegistration () {
+    await this.props.completeRegistration(this.props.domainName);
+    await this.props.wait();
+    this.showConfirmTransactionPage(() => this.afterConfirmRegistration());
+  }
+
+  viewDomainDetails () {
+    this.props.viewDomainDetails(this.props.domainName);
+  }
+
+  afterRegistration () {
+    this.props.showThis({
+      ...this.props,
+      currentStep: 'registered',
+    });
+    this.changeDomainStatusAndStore();
+  }
+
+  afterConfirmRegistration () {
+    this.props.showThis({
+      ...this.props,
+      currentStep: 'registerConfirmation',
+    });
+  }
+
+  showConfirmTransactionPage (callback, payload) {
+    this.props.getUnapprovedTransactions()
+      .then(latestTransaction => {
+        this.props.showTransactionConfirmPage({
+          id: latestTransaction.id,
+          unapprovedTransactions: latestTransaction,
+          afterApproval: {
+            action: (payload) => {
+              if (callback) {
```

Comment:
Can we avoid this payload check? If we execute callback(payload) and payload is undefined or falsy, then the callback function should handle it exactly the same as without arguments, right?

Reply:
Yeah i think you are right.